### ISSUE TITLE
Follow-up changes for System.ComponentModel

### ISSUE THREAD

**Issue by AlexGhiondea** [dotnet/corefx#12212]
- [x] Once #11809 is checked in
  - [x] Make Component derive from MarshalByRefObject
- [x] Once #12209 is checked in
  - [x] Update the attributes in the ComponentModel namespace to implement those methods. As part of that change, we need to revert the change that introduced [this workaround](https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentModelExtensions.cs)
- [x] Once #11769 is done:
  - [x] Re-introduce the parameter-less constructor on ComponentResourceManager
  - [x] Remove the implementation of ResourceSet from misc.cs
- [x] Once #11769 is done, fixup the following stubs inside the misc.cs file.
  - [x] ResourceManager_MainAssembly
  - [x] ResourceManager_GetNeutralResourceLanguage
  - [x] ResourceManager_IgnoreCase
  - [x] ResourceManager_GetResourceSet
- [x] Once #11655 is done, fixup the following stubs inside the misc.cs file:
  - [x] Assembly_EscapedCodeBase
- [x] Once #12211 is done, fixup the following stubs inside the misc.cs file:
  - [x] AppDomain_CurrentDomain_SetupInformation_LicenseFile
  - [x] AppDomain_CurrentDomain_GetAssemblies
- [x] Remove dependency to System.Net.WebClient from the DesigntimeLicenseContext type

**Comment by AlexGhiondea** [comment]
Filed a new issue to track the remaining work.
https://github.com/dotnet/corefx/issues/14695
